cogip/sensor.py

```diff
@@ -8,46 +9,61 @@
 
 class Sensor(QtCore.QObject):
 
+    # Class attribute recording all entities that should be detected
+    obstacles = []
+
+    # Class attribute recording all sensors
+    # Each added obstacle must be registered in all sensors
+    all_sensors = []
+
     def __init__(
             self,
             asset_entity: AssetEntity,
             name: str,
             origin_x: int,
             origin_y: int,
+            origin_z: int,
+            direction_x: int,
+            direction_y: int,
+            direction_z: int,
             impact_radius: float = 50,
-            impact_color: QtCore.Qt.red = QtCore.Qt.red):
+            impact_color: QtCore.Qt.GlobalColor = QtCore.Qt.red):
         super(Sensor, self).__init__()
 
+        Sensor.all_sensors.append(self)
+
         self.asset_entity = asset_entity
         self.name = name
         self.distance = None
 
-        direction_x = 1.0 if origin_x > 0 else -1.0 if origin_x < 0 else 0.0
-        direction_y = 1.0 if origin_y > 0 else -1.0 if origin_y < 0 else 0.0
-
         self.ray_caster = Qt3DRender.QRayCaster()
-        self.ray_caster.setEnabled(True)
-        self.ray_caster.setLength(0)  # Infinite
+        self.ray_caster.setEnabled(False)
+        self.ray_caster.setLength(2000)  # Infinite
```

Comment:
It seems no more infinite.
Why 2000 ? Can you create a constant to define this ?

Reply:
This was a test. Reverted.